source/core/slang-io.cpp

```diff
@@ -286,7 +286,23 @@ namespace Slang
             // If ioBuilder doesn't end in a delimiter, add one
             if (!isDelimiter(ioBuilder[ioBuilder.getLength() - 1]))
             {
-                ioBuilder.append(kPathDelimiter);
+                // Determine the preferred delimiter to use based on existing path.
+                char preferedDelimiter = kOSCanonicalPathDelimiter;
+                if (kOSAlternativePathDelimiter != preferedDelimiter)
+                {
+                    // If we found the existing path uses the alternative delimiter, we will
+                    // use that instead of the canonical one.
+                    constexpr Index kMaxDelimiterSearchRange = 32;
+                    for (Index i = 0; i < Math::Min(kMaxDelimiterSearchRange, ioBuilder.getLength()); i++)
+                    {
+                        if (ioBuilder[i] == kOSAlternativePathDelimiter)
+                        {
+                            preferedDelimiter = kOSAlternativePathDelimiter;
+                            break;
+                        }
```

Comment:
I suppose I'm not 100% clear on the desire to use an alternate delimiter if it's already seen? It would be a little odd to see a path with different delimiters. In any kind of lookup/hashing paths should be in one style (presumably the normal delimiter).

Reply:
This is to preserve the current compiler behavior as much as possible. We can always revise this and perform canonicalization upon combine, but I think it is safer to just preserve the existing style.

Reply:
For example, if the combine("https://a", "b") should result "https://a/b") even on windows. The easiest way to make sure such case to work is to detect and use existing delimiters.